spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -117,27 +119,29 @@ public InternalRow[] call(InternalRow args) {
           }
 
           String where = args.isNullAt(4) ? null : args.getString(4);
-
-          action = checkAndApplyFilter(action, where, quotedFullIdentifier);
+          if (where != null) {
+            Expression icebergExpression = filter(where, quotedFullIdentifier);
+            action = action.filter(icebergExpression);
+          }
 
           RewriteDataFiles.Result result = action.execute();
-
```

Comment:
nit: unnecessary change

Reply:
yes. you are right. let me change my code.